it/endpoints/CreateTestBusinessControllerISpec.scala

```diff
@@ -79,17 +80,26 @@ class CreateTestBusinessControllerISpec extends UnitSpec with IntegrationBaseSpe
 
   "Calling the create Business endpoint" should {
     "return a 201 status code" when {
-      "a valid request is made" in new Test {
+      "a valid request is made with a self-employment business" in new Test {
+        val response: WSResponse = await(request().post(MinimalCreateTestBusinessRequest.SelfEmployment.mtdBusinessJson))
 
-        val response: WSResponse = await(request().post(MinimalCreateTestBusinessRequest.mtdBusinessJson))
+        response.status shouldBe CREATED
+        response.header("X-CorrelationId") should not be empty
+        response.json shouldBe Json.parse(expectedResponseBody)
+      }
+
+      "a valid request is made with a uk-property business" in new Test {
+        val response: WSResponse = await(request().post(MinimalCreateTestBusinessRequest.UkProperty.mtdBusinessJson))
 
         response.status shouldBe CREATED
         response.header("X-CorrelationId") should not be empty
         response.json shouldBe Json.parse(expectedResponseBody)
       }
     }
+
     "return validation error according to spec" when {
-      def validationErrorTest(requestNino: String, requestBody: JsObject, expectedStatus: Int, expectedBody: MtdError): Unit = {
+      def validationErrorTest(requestNino: String, requestBody: JsObject, expectedStatus: Int, expectedBody: MtdError)
+                             (implicit pos: Position): Unit = {
```

Comment:
A lot of the test refactoring had to do with me adding the implicit Position because it was annoying to chase the failing parameters in one of these files.

As a rule, one should add an implicit Position when the body of the method is not as important as where it is called.  
e.g.
- the function/method is a general-purpose test utility.
- the function contains very obvious assertions that are called with less obvious parameters in multiple places.

It doesn't change the stack traces. But it changes the reported line of test failures to the line where the implicit is first generated. In this case, the test failures will first point to where the test method is called, rather than the single shouldBe assertion that failed. Makes it easier to jump to the parameters that caused the failure.

For general-purpose testing utilities, it's useful because it points the user to where the utility was called instead of some unimportant line in the called utility.

The stack traces remain the same, so no information is completely lost. This is most useful in IntelliJ when you double-click on a failed test outcome.